crates/ruff/src/rules/flake8_bugbear/snapshots/ruff__rules__flake8_bugbear__tests__B006_B006_B008.py.snap

```diff
@@ -1,113 +1,344 @@
    |                         ^^^^^^^^^ B006
 64 |     ...
    |
+   = help: Replace mutable data structure with `None` in argument default and replace it with data structure inside the function if still `None`
 
-B006_B008.py:67:30: B006 Do not use mutable data structures for argument defaults
+ℹ Possible fix
+60 60 | # Flag mutable literals/comprehensions
+61 61 | 
+62 62 | 
+63    |-def this_is_wrong(value=[1, 2, 3]):
+   63 |+def this_is_wrong(value=None):
+   64 |+    if value is None:
+   65 |+        value = [1, 2, 3]
+64 66 |     ...
+65 67 | 
+66 68 | 
+
+B006_B008.py:67:30: B006 [*] Do not use mutable data structures for argument defaults
    |
 67 | def this_is_also_wrong(value={}):
    |                              ^^ B006
 68 |     ...
    |
+   = help: Replace mutable data structure with `None` in argument default and replace it with data structure inside the function if still `None`
+
+ℹ Possible fix
+64 64 |     ...
+65 65 | 
+66 66 | 
+67    |-def this_is_also_wrong(value={}):
+   67 |+def this_is_also_wrong(value=None):
+   68 |+    if value is None:
+   69 |+        value = {}
+68 70 |     ...
+69 71 | 
+70 72 | 
 
-B006_B008.py:71:20: B006 Do not use mutable data structures for argument defaults
+B006_B008.py:73:52: B006 [*] Do not use mutable data structures for argument defaults
    |
-71 | def and_this(value=set()):
-   |                    ^^^^^ B006
-72 |     ...
+71 | class Foo:
+72 |     @staticmethod
+73 |     def this_is_also_wrong_and_more_indented(value={}):
+   |                                                    ^^ B006
+74 |         pass
    |
+   = help: Replace mutable data structure with `None` in argument default and replace it with data structure inside the function if still `None`
 
-B006_B008.py:75:20: B006 Do not use mutable data structures for argument defaults
+ℹ Possible fix
+70 70 | 
+71 71 | class Foo:
+72 72 |     @staticmethod
+73    |-    def this_is_also_wrong_and_more_indented(value={}):
+   73 |+    def this_is_also_wrong_and_more_indented(value=None):
+   74 |+        if value is None:
+   75 |+            value = {}
+74 76 |         pass
+75 77 | 
+76 78 | 
+
+B006_B008.py:77:31: B006 [*] Do not use mutable data structures for argument defaults
    |
-75 | def this_too(value=collections.OrderedDict()):
-   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^ B006
-76 |     ...
+77 |   def multiline_arg_wrong(value={
+   |  _______________________________^
+78 | | 
+79 | | }):
+   | |_^ B006
+80 |       ...
    |
+   = help: Replace mutable data structure with `None` in argument default and replace it with data structure inside the function if still `None`
 
-B006_B008.py:79:32: B006 Do not use mutable data structures for argument defaults
+ℹ Possible fix
+74 74 |         pass
+75 75 | 
+76 76 | 
+77    |-def multiline_arg_wrong(value={
+78    |-
+79    |-}):
+   77 |+def multiline_arg_wrong(value=None):
+   78 |+    if value is None:
+   79 |+        value = {}
```

Comment:
it's odd that this is collapsed here. what happens if the content is mandatory multi line, e.g. with a comment in the parentheses?

Reply:
We end up deleting the comment. Similar thing popped up in a [different PR](https://github.com/astral-sh/ruff/pull/5121#discussion_r1237755195) where I was told most Ruff fixes ignore comments and not doing so requires libCST.  @MichaReiser does the same principle apply here?

Reply:
Or if the dictionary is pretty long:
```python
def multiline_arg(
    arg={
        "key1": "value",
        "key2": "value",
        "key3": "value",
        "key4": "value",
        "key5": "value",
        "key6": "value",
    }
):
    pass
```

It's getting collapsed as well:
```diff
--- /Users/dhruv/playground/ruff/src/play.py
+++ /Users/dhruv/playground/ruff/src/play.py
@@ -1,11 +1,6 @@
 def multiline_arg(
-    arg={
-        "key1": "value",
-        "key2": "value",
-        "key3": "value",
-        "key4": "value",
-        "key5": "value",
-        "key6": "value",
-    }
+    arg=None
 ):
+    if arg is None:
+        arg = {"key1": "value", "key2": "value", "key3": "value", "key4": "value", "key5": "value", "key6": "value"}
     pass
```

Reply:
I think that's fine for a first version. We can either try to improve this in a follow up PR, wait for Ruff's formatter to automatically fix it, or leave it as is.